wfe/wfe.go

```diff
@@ -971,6 +998,76 @@ func (wfe *WebFrontEndImpl) UpdateAccount(
 	}
 }
 
+func (wfe *WebFrontEndImpl) ListOrders(
+	ctx context.Context,
+	response http.ResponseWriter,
+	request *http.Request) {
+	postData, prob := wfe.verifyPOST(request, wfe.lookupJWK)
+	if prob != nil {
+		wfe.sendError(prob, response)
+		return
+	}
+
+	existingAcct, prob := wfe.validPOSTAsGET(postData)
+	if prob != nil {
+		wfe.sendError(prob, response)
+		return
+	}
+
+	var page int
+	accountPageStr := strings.TrimPrefix(request.URL.Path, ordersPath)
+	accountPageSplit := strings.SplitN(accountPageStr, "/page/", 2)
+	if len(accountPageSplit) == 0 || accountPageSplit[0] != existingAcct.ID {
+		wfe.sendError(acme.NotFoundProblem("Wrong account ID"), response)
+		return
+	}
+	if len(accountPageSplit) == 2 {
+		no, err := strconv.Atoi(accountPageSplit[1])
+		if err != nil || no < 2 {
+			wfe.sendError(acme.NotFoundProblem("Invalid page number"), response)
+			return
+		}
+		page = no - 1
+	}
+	orders := wfe.db.GetOrdersByAccountID(existingAcct.ID)
+	filteredOrders := make([]*core.Order, 0, len(orders))
```

Comment:
I think if you switch this to:
```
var filteredOrders []*core.Order
for _, order := range orders {
  ...
  filteredOrders = append(filteredOrders, order)
}
```
then a `nil` result from `wfe.db.GetOrdersByAccountID` would be handled fine. You would end up with an empty `filteredOrders` slice and could check if `len == 0` and exit early.

Reply:
I think there's no code change needed to accomodate for `nil`. Both `len()` and `range` seem to work fine when `orders` is `nil`.

Reply:
Just tested it, works fine.

Reply:
Good call! Thanks @felixfontein